test/interpolating_tests.jl

```diff
@@ -0,0 +1,70 @@
+using DifferentialEquations, SciMLSensitivity, Zygote
+using QuadGK
+using ForwardDiff
+using Interpolations
+using Test
+using FastGaussQuadrature
+
+# function for computing vector-jacobian products using Zygote
+function vjp(func, eval_pt, vec_mul)
+    _, func_pullback = pullback(func, eval_pt)
+    vjp_result = func_pullback(vec_mul)
+    return vjp_result
+end
+
+# loss function
+function g(u,p,t)
+    return sum((u .- 1.0).^2)
+end
+
+function lotka_volterra(u, p, t)
+    x, y = u
+    α, β, δ, γ = p
+    dx = α * x - β * x * y
+    dy = -δ * y + γ * x * y
+    return [dx,dy]
+end
+
+function adjoint(u, p, t, sol)
+    return -vjp((x)->lotka_volterra(x,p,t), sol(t), u)[1] - Zygote.gradient((x)->g(x,p,t), sol(t))[1]
+end
+
+u0 = [1.0, 1.0] #initial condition
+tspan = (0.0, 10.0) #simulation time
+p = [1.5, 1.0, 3.0, 1.0] # Lotka-Volterra parameters
+prob = ODEProblem(lotka_volterra, u0, tspan, p)
+sol = solve(prob, Tsit5(), abstol = 1e-14, reltol = 1e-14)
+
+# total loss functional
+function G(p)
+    tmp_prob = remake(prob, p = p)
+    sol = solve(tmp_prob, Tsit5(), abstol = 1e-14, reltol = 1e-14)
+    res, = quadgk((t) -> g(sol(t),p,t), tspan[1], tspan[2], atol = 1e-14, rtol = 1e-10)
+    return res
+end
+
+dGdp_ForwardDiff = ForwardDiff.gradient(G, p) 
+
+integrand_values = IntegrandValues(Vector{Float64})
+function callback_saving(u,t,integrator,sol)
+    temp = sol(t)
+    return vjp((x)->lotka_volterra(temp,x,t),integrator.p,u)[1]
+end
+gp, ws = gausslegendre(5) #generate Gaussian quadrature nodes
```

Comment:
pre-compute these in a table up to 7, so 10*2 + 1 = 21 which covers up to 21st order methods which is more than enough. Then in the method choose which one from the table based on `DiffEqBase.alg_order(integrator.alg)` so that gausspoints * 2 + 1 > DiffEqBase.alg_order(integrator.alg), in other words, div(DiffEqBase.alg_order(integrator.alg) - 1,2) rounding up

Reply:
Where would this go? I imagine we'll use this callback to implement a new sensitivity algorithm in SciMLSensitivity.jl. Would it make sense to precompute the table there?

Reply:
This functionality is useful even completely outside of the context of adjoints, so I wouldn't worry about that. That would just go into this package. It would just be a constant array of arrays where `weights = [gausslegendre(i) for i in 1:10]` (handle gp vs ws of course). Build it and then `@show weights` to get the Julia definition of it and then copy-paste `weights = ...` to the file, along with a comment on how it was generated. Then `weights[i]` is all you need.

Reply:
fixed this! had to add `OrdinaryDiffEq.jl` to the dependencies (for `alg_order`). Also the wikipedia page for gaussian quadrate has degree=2n-1, so I used this instead of 2n+1

Reply:
> I had to add OrdinaryDiffEq.jl to the dependencies (for alg_order)

You don't need OrdinaryDiffEq, you just need SciMLBase https://github.com/SciML/SciMLBase.jl/blob/master/src/alg_traits.jl#L137